samples/cpp/prompt_lookup_decoding_lm/prompt_lookup_decoding_lm.cpp

```diff
@@ -1,338 +1,47 @@
-        input_ids.set_shape({BATCH_SIZE, candidates_size + 1});
-        input_ids.data<int64_t>()[0] = first_token;
-        std::copy_n(candidates.begin(), candidates_size, input_ids.data<int64_t>() + 1);
-
-        attention_mask.set_shape({BATCH_SIZE, seq_len + candidates_size + 1});
-        std::fill_n(attention_mask.data<int64_t>(), attention_mask.get_size(), 1);
-
-        position_ids.set_shape({BATCH_SIZE, candidates_size + 1});
-        std::iota(position_ids.data<int64_t>(), position_ids.data<int64_t>() + position_ids.get_size(), seq_len);
-
-        model.infer();
-
-        data_logits = logits.data<float>();  // [BATCH_SIZE, 1 + candidates_size, vocab_size]
-
-        // 1. accept current out token (if not eos)
-        // 2. check if it matches appropriate candidate
-        //      2.1 if it's match, continue - accept next token
-        //      2.2 it it's mismatch, stop iteration but still accept current token as it was last token generated by
-        //      model from a valid sequence.
-        size_t accepted_tokens_number = 0;
-        for (size_t i = 0; i < candidates_size + 1; i++) {
-            auto start = data_logits + vocab_size * i;
-            auto stop = data_logits + vocab_size * (i + 1);
-            out_token = std::max_element(start, stop) - start;
-
-            if (out_token == EOS_TOKEN) {
-                break;
-            }
-
-            text_streamer.put(out_token);
-            full_input_ids.push_back(out_token);
-            accepted_tokens_number++;
-
-            if (i == candidates_size || out_token != candidates[i]) {
-                break;
-            }
-        }
-
-        if (accepted_tokens_number > 0) {
-            candidateGenerator.update_candidate_strategy(accepted_tokens_number - 1);
-        }
-
-        // After the inference request, key/values have shape [BATCH_SIZE, seq_len + candidates_size, vocab_size].
-        // Increment the sequence length by the number of matched tokens, and
-        // trim the KV cache to match the new sequence length.
-        seq_len += accepted_tokens_number;
-        update_kv_cache(model, seq_len_axis, seq_len);
-
-        first_token = out_token;
-    }
-
-    text_streamer.end();
-    // Model is stateful which means that context (kv-cache) which belongs to a particular
-    // text sequence is accumulated inside the model during the generation loop above.
-    // This context should be reset before processing the next text sequence.
-    // While it is not required to reset context in this sample as only one sequence is processed,
-    // it is called for education purposes:
-    model.reset_state();
+    ov::genai::GenerationConfig config;
+    config.max_new_tokens = 100;
+    // Speculative decoding generation parameters like `num_assistant_tokens` and `assistant_confidence_threshold` are mutually excluded
+    // add parameter to enable speculative decoding to generate `num_assistant_tokens` candidates by draft_model per iteration
```

Comment:
looks like comments need to be updated as they mention draft model.

Here as well
https://github.com/openvinotoolkit/openvino.genai/blob/7d8912ff9df9bcfacf0044d108963cb7618bff69/src/cpp/include/openvino/genai/generation_config.hpp#L76

It would be also good to update readme to describe parameters in more details. For unexperienced users it can be unclear what parameters affect prompt look-up